types/ensure/package.pp

```diff
@@ -0,0 +1,18 @@
+# https://docs.puppet.com/puppet/latest/type.html#package-attribute-ensure
+# ensure
+#
+# (Property: This attribute represents concrete state on the target system.)
+#
+# What state the package should be in. On packaging systems that can retrieve
+# new packages on their own, you can choose which package to retrieve by
+# specifying a version number or latest as the ensure value. On packaging
+# systems that manage configuration files separately from
+# rmaystem files, you can uninstall config files by specifying purged as the ensure value. This defaults to installed.
+#
+# Version numbers must match the full version to install, including release if
+# the provider uses a release moniker. Ranges or semver patterns are not accepted
+# except for the gem package provider. For example, to install the bash package
+# from the rpm bash-4.1.2-29.el6.x86_64.rpm, use the string '4.1.2-29.el6'.
+#
+# Valid values are present (also called installed), absent, purged, held, latest. Values can match /./.
+type Tea::Ensure::Package = Enum['present', 'installed', 'absent', 'purged', 'held', 'latest']
```

Comment:
A package can also be an exact version number which means `String` is the only valid type. I actually wouldn't mind a type wrapping `String[1]` for that just to indicate how the variable is used though. IMHO that also makes sense in stdlib.

Reply:
Yes, I was actually doing the same when I made the PR, but decided against it. My reasoning was that adding `String[1]` basically breaks the type validation, as anything would be allowed.

Maybe it'd be better to have a strict (only `Enum['present', 'installed', 'absent', 'purged', 'held', 'latest']`) and a non-strict type (where strings are also allowed)?